pkg/es/config/config.go

```diff
@@ -188,10 +202,62 @@ func (c *Configuration) GetTagDotReplacement() string {
 }
 
 // GetConfigs wraps the configs to feed to the ElasticSearch client init
-func (c *Configuration) GetConfigs() []elastic.ClientOptionFunc {
-	options := make([]elastic.ClientOptionFunc, 3)
-	options[0] = elastic.SetURL(c.Servers...)
-	options[1] = elastic.SetBasicAuth(c.Username, c.Password)
-	options[2] = elastic.SetSniff(c.Sniffer)
+func (c *Configuration) GetConfigs(logger *zap.Logger) []elastic.ClientOptionFunc {
+	options := make([]elastic.ClientOptionFunc, 0)
+	options = append(options, elastic.SetURL(c.Servers...), elastic.SetSniff(c.Sniffer))
+	if c.TLS.Enabled {
+		ctlsConfig, err := c.TLS.createTLSConfig(logger)
+		if err != nil {
+			return nil
+		}
+		httpClient := &http.Client{
+			Timeout: c.Timeout,
+			Transport: &http.Transport{
+				TLSClientConfig: ctlsConfig,
+			},
+		}
+		options = append(options, elastic.SetHttpClient(httpClient))
+	} else {
+		options = append(options, elastic.SetBasicAuth(c.Username, c.Password))
+	}
 	return options
 }
+
+// createTLSConfig creates TLS Configuration to connect with ES Cluster.
+func (tlsConfig *TLSConfig) createTLSConfig(logger *zap.Logger) (*tls.Config, error) {
+	rootCerts, err := tlsConfig.loadCertificate()
+	if err != nil {
+		logger.Fatal("Couldn't load root certificate", zap.Error(err))
+	}
+	if len(tlsConfig.CertPath) > 0 && len(tlsConfig.KeyPath) > 0 {
```

Comment:
why is this check needed?  If it fails you return nil, so shouldn't it be an error to set `c.TLS.Enabled` and not provide the other 3 params?